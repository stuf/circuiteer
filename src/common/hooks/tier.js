import * as L from 'partial.lenses';
import { useSelector } from 'react-redux';

export function useCurrentTier() {
  const current = useSelector(L.get(['tier', 'current']));

  return current;
}
