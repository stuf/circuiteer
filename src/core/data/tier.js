export const Tier = {
  OTHER: -1,
  SMALL: 1,
  MEDIUM: 2,
  LARGE: 3,
  XLARGE: 4,
};

export const TierName = {
  [Tier.OTHER]: 'other',
  [Tier.SMALL]: 'small',
  [Tier.MEDIUM]: 'medium',
  [Tier.LARGE]: 'large',
  [Tier.XLARGE]: 'extraLarge',
};
