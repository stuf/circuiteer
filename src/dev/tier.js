const defaults = [
  { id: 1, name: 'small' },
  { id: 2, name: 'medium' },
  { id: 3, name: 'large' },
  { id: 4, name: 'extraLarge' },
  { id: -1, name: 'other' },
];

export default defaults;
