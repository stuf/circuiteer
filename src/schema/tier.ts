export enum Tier {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large',
  EXTRA_LARGE = 'extraLarge',
}
