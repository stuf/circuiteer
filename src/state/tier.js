import * as L from 'partial.lenses';
import { createSlice, original } from '@reduxjs/toolkit';

import { createPrefixedActionCreator } from 'common/util';

const name = 'tier';
const createAction = createPrefixedActionCreator(name);

const initialState = {
  tiers: [],
  current: 1,
};

const stateL = L.pickIn({
  tiers: L.define([]),
  current: L.define(null),
});

//

export const setTiers = createAction('setTiers');
export const setCurrentTier = createAction('setCurrentTier');

//

const slice = createSlice({
  name,
  initialState,
  reducers: {},
  extraReducers: builder =>
    builder
      .addCase(setTiers, (s, a) =>
        L.set([stateL, 'tiers'], a.payload, original(s)),
      )
      .addCase(setCurrentTier, (s, a) =>
        L.set([stateL, 'current', L.rewrite(v => +v)], a.payload, original(s)),
      ),
});

const reducer = slice.reducer;

export default reducer;
