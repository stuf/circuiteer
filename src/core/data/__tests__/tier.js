import { Tier } from '../tier';

test('Tier', () => {
  expect(Tier).toMatchSnapshot();
});
