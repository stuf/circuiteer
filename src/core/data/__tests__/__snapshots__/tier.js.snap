// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tier 1`] = `
Object {
  "LARGE": 3,
  "MEDIUM": 2,
  "OTHER": -1,
  "SMALL": 1,
  "XLARGE": 4,
}
`;
